app/controllers/registrations_controller.rb

```diff
@@ -0,0 +1,34 @@
+class RegistrationsController < ApplicationController
+  skip_before_action :authorize
+
+  def create
+    @user = User.new(user_params)
+
+    if @user.save
+      redirect_to login_path, notice: t('notice.registration.signup_success')
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def verify_email
+    if current_user_by_confirmation_token
```

Comment:
load user here if not resused or redirected

Reply:
use before_action for loading user and redirect if not valid